tests/integration/cloudformation/nextgen/test_deploy.py

```diff
@@ -0,0 +1,43 @@
+from typing import TypeVar
+
+import pytest
+
+from localstack.services.opensearch.resource_providers.domain import OpenSearchDomainAllProperties
+from localstack.services.ssm.resource_providers.parameter import SSMParameterAllProperties
+from localstack.utils.strings import short_uid
+
+Properties = TypeVar("Properties")
+
+
+@pytest.mark.parametrize(
+    "type_name,props",
+    [
+        (
+            "AWS::OpenSearchService::Domain",
+            OpenSearchDomainAllProperties(DomainName=f"domain-{short_uid()}"),
+        ),
+        (
+            "AWS::SSM::Parameter",
+            SSMParameterAllProperties(Type="String", Value=f"value-{short_uid()}"),
+        ),
+        # TODO: Add your resource definitions here!
+    ],
+    ids=["opensearch-domain", "ssm-parameter"],
+)
+@pytest.mark.skip(reason="Example")
+def test_roundtrip(type_name, props, perform_cfn_operation):
+    # deploy
+    event = perform_cfn_operation(
+        logical_resource_id="MyResource",
+        resource_type=type_name,
+        action="Add",
+        resource_props=props,
+    )
+
```

Comment:
I feel like this kind of test would be missing some validation in the middle here to check if the resource has actually been "deployed" :thinking:

Reply:
I agree, but at the moment we aren't asking for the `read` handler to be implemented, so writing a generic "is the resource there" assertion is not currently possible.

This file was mostly for me to help drive out the outer deploy functionality so I'm not too bothered right now, and we can revisit this later.